### PR TITLE
Update libssh2 from 1.9.0 to 1.10.0

### PR DESCRIPTION
**Pull request by cbouss**
# Changes

Changes:
- Update version from `1.9.0` to `1.10.0`.
- Update sha256
- Added doc_url and doc_source_url
- Remove patch for CVE-2019-17498 applicable up to 1.9.0 only.
- Remove missing test files applicable to 1.9.0 only.

# Review Information

## Source

https://github.com/libssh2/libssh2/tree/libssh2-1.10.0

## License

https://github.com/libssh2/libssh2/blob/libssh2-1.10.0/COPYING

## Upstream Changes

https://github.com/libssh2/libssh2/blob/libssh2-1.10.0/RELEASE-NOTES

## Issues

https://github.com/libssh2/libssh2/issues

## Pins and Requireds

NA

## Testing

Existing tests passed.

# Closing Comments

libssh2 udpated to 1.10.0

### CODE REVIEW

**Comment by cbouss** [comment]
Building on win64: https://concourse.build.corp.continuum.io/teams/main/pipelines/cb_libssh/jobs/libssh2-1.10.0-on-winbuilder/builds/1

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
-    # https://nvd.nist.gov/vuln/detail/CVE-2019-17498
-    # patch from the Fedora project
-    # https://src.fedoraproject.org/rpms/libssh2/c/41525baf
-    # merged into upstream in with small modification in:
-    # https://github.com/libssh2/libssh2/pull/402
-    - 0001-libssh2-1.9.0-CVE-2019-17498.patch
+  sha256: 2d64e90f3ded394b91d3a2e774ca203a4179f69aebee03003e5a6fa621e41d51
 
 build:
   number: 1
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]: The build number should be `0` if we update to a new version

File: recipe/build.sh [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
 # We use a repackaged cmake from elsewhere to break a build cycle.
 export PATH=${PREFIX}/cmake-bin/bin:${PATH}
 
 if [[ $target_platform =~ linux.* ]]; then
   export LDFLAGS="$LDFLAGS -Wl,-rpath-link,$PREFIX/lib"
 fi
 
+# linux-aarch64 activations fails to set `ar` tool. This can be
+# removed when ctng-compiler-activation is corrected.
+if [[ "${target_platform}" == linux-aarch64 ]]; then
```
Comment (katietz) [inline recipe/build.sh:9]: Have you filed a Jira ticket for this? if not, please do so

File: recipe/meta.yaml [hunk 3]
```diff
@@ -27,10 +19,7 @@ requirements:
     - make   # [unix]
     # This breaks a dependency cycle:
     # curl->libssh2->cmake->curl
-    - cmake-binary  # [not (osx and arm64)]
```
Comment (katietz) [inline recipe/meta.yaml:30]: do not remove thos cmake packages.  They are there to allow building without cyclic dependency in cmake.

File: recipe/meta.yaml [hunk 4]
```diff
@@ -27,10 +19,7 @@ requirements:
...
-    - cmake-no-system         # [osx and arm64]
-    - patch       # [not win]
-    - m2-patch    # [win]
+    - cmake-no-system
```
Comment (katietz) [inline recipe/meta.yaml:22]: is this package available for all architectures? Also for windows?

File: recipe/build.sh [hunk 5]
```diff
@@ -1,21 +1,25 @@
 #!/bin/bash
 
-# copy files which are missing from the release tarball
-# see: https://github.com/libssh2/libssh2/issues/379
-# TODO: remove this in the 1.9.1 or later releases
-cp ${RECIPE_DIR}/missing_files/*.c tests/
-
 # We use a repackaged cmake from elsewhere to break a build cycle.
 export PATH=${PREFIX}/cmake-bin/bin:${PATH}
```
Comment (katietz) [inline recipe/build.sh]: see this seems to be an line related to removed cmake-binary package

**Review by katietz — CHANGES_REQUESTED** [review]

File: recipe/build.sh (hunk 2 above)
Comment (cbouss) [reply recipe/build.sh:9]: Did so here: https://anaconda.atlassian.net/browse/DSNC-4678
I tried a fix here, but it needs more work: https://github.com/AnacondaRecipes/ctng-compiler-activation-feedstock/pull/3

File: recipe/meta.yaml (hunk 3 above)
Comment (cbouss) [reply recipe/meta.yaml:30]: This is fulfilled by cmake-no-system, which is available for all architectures.

File: recipe/meta.yaml (hunk 4 above)
Comment (cbouss) [reply recipe/meta.yaml:22]: Yes it is available for all architectures.

File: recipe/build.sh (hunk 5 above)
Comment (cbouss) [reply recipe/build.sh]: These files used to be missing from libssh2 1.9.0 release. They are present in 1.10.0 release.

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml]: done

File: recipe/meta.yaml (hunk 3 above)
Comment (katietz) [reply recipe/meta.yaml:30]: thanks

File: recipe/meta.yaml (hunk 4 above)
Comment (katietz) [reply recipe/meta.yaml:22]: ok good, then I am fine with that

File: recipe/build.sh (hunk 5 above)
Comment (katietz) [reply recipe/build.sh]: well, that path variable looks to me superflous, isn't it?

Comment (katietz) [reply recipe/build.sh]: by taking a look to cmake-no-system-feedstock (which is used as build dependency) adding cmake-bin as path shouldn't be necessary here

Comment (cbouss) [reply recipe/build.sh]: right, I removed the path setting.

**Review by katietz — APPROVED** [review]
thank you!

**Pull request merged**